common/src/test/java/com/revenuecat/purchases/common/BackendTest.kt

```diff
@@ -277,6 +278,92 @@ class BackendTest {
         }
     }
 
+    @Test
+    fun `postReceipt passes pricing phases as maps in body`() {
+        val purchaseOption = storeProduct.purchaseOptions[0]
+        val receiptInfo = ReceiptInfo(
+            productIDs = productIDs,
+            storeProduct = storeProduct,
+            purchaseOptionId = purchaseOption.id
+        )
+
+        mockPostReceiptResponseAndPost(
+            backend,
+            isRestore = false,
+            observerMode = false,
+            receiptInfo = receiptInfo,
+            storeAppUserID = null
+        )
+
+        val expectedPricingPhases = receiptInfo.pricingPhases
+        val expectedPricingPhasesMap = expectedPricingPhases?.map { it.toMap() }
+        assertThat(requestBodySlot.isCaptured).isTrue
+        assertThat(requestBodySlot.captured.keys).contains("pricing_phases")
+        assertThat(requestBodySlot.captured["pricing_phases"]).isEqualTo(expectedPricingPhasesMap)
+
+        expectedPricingPhases?.forEachIndexed { index, pricingPhase ->
+            val mappedPricingPhase = expectedPricingPhasesMap?.get(index)
+            assertThat(mappedPricingPhase).isNotNull.withFailMessage(
+                "there should be a mapped version for every pricingPhase"
+            )
+            assertThat(mappedPricingPhase?.get("billingPeriod")).isEqualTo(pricingPhase.billingPeriod)
+            assertThat(mappedPricingPhase?.get("billingCycleCount")).isEqualTo(pricingPhase.billingCycleCount)
+            assertThat(mappedPricingPhase?.get("formattedPrice")).isEqualTo(pricingPhase.formattedPrice)
+            assertThat(mappedPricingPhase?.get("priceCurrencyCode")).isEqualTo(pricingPhase.priceCurrencyCode)
+            assertThat(mappedPricingPhase?.get("recurrenceMode")).isEqualTo(pricingPhase.recurrenceMode.identifier)
```

Comment:
What do you think about adding a check for the size of the map? That way if we add a new property we don't forget to map it

Reply:
hmmm can you elaborate on what that would look like? i could use reflection to get the number of fields in PricingPhase and make sure the map has the same? otherwise not sure how it'd catch a new field missing

Reply:
nono, it was a good idea.. maybe we can think about how that for the future. i'm sure there are other places we could use a test like that